src/meta/src/stream/sink_manager.rs

```diff
@@ -0,0 +1,179 @@
+    S: MetaStore,
+{
+    const SINK_TICK_INTERVAL: Duration = Duration::from_secs(10);
+
+    pub async fn new(
+        env: MetaSrvEnv<S>,
+        cluster_manager: ClusterManagerRef<S>,
+        fragment_manager: FragmentManagerRef<S>,
+    ) -> Result<Self> {
+        let managed_sinks = HashMap::new();
+        let core = Arc::new(Mutex::new(SinkManagerCore::new(
+            fragment_manager,
+            managed_sinks,
+        )));
+
+        Ok(Self {
+            env,
+            cluster_manager,
+            core,
+        })
+    }
+
+    async fn all_stream_clients(&self) -> Result<impl Iterator<Item = StreamClient>> {
+        let all_compute_nodes = self
+            .cluster_manager
+            .list_worker_node(WorkerType::ComputeNode, Some(Running))
+            .await;
+
+        let all_stream_clients = try_join_all(
+            all_compute_nodes
+                .iter()
+                .map(|worker| self.env.stream_client_pool().get(worker)),
+        )
+        .await?
+        .into_iter();
+
+        Ok(all_stream_clients)
+    }
+
+    /// Broadcast the create sink request to all compute nodes.
+    pub async fn create_sink(&self, sink: &Sink) -> Result<()> {
+        // This scope guard does clean up jobs ASYNCHRONOUSLY before Err returns.
+        // It MUST be cleared before Ok returns.
+        let mut revert_funcs = scopeguard::guard(
+            vec![],
+            |revert_funcs: Vec<futures::future::BoxFuture<()>>| {
+                tokio::spawn(async move {
+                    for revert_func in revert_funcs {
+                        revert_func.await;
+                    }
+                });
+            },
+        );
+
+        let futures = self
+            .all_stream_clients()
+            .await?
+            .into_iter()
+            .map(|mut client| {
+                let request = ComputeNodeCreateSinkRequest {
+                    sink: Some(sink.clone()),
+                };
+                async move { client.create_sink(request).await.map_err(RwError::from) }
+            });
+
+        // ignore response body, always none
+        let _ = try_join_all(futures).await?;
+
+        let core = self.core.lock().await;
+        if core.managed_sinks.contains_key(&sink.get_id()) {
+            log::warn!("sink {} already registered", sink.get_id());
+            revert_funcs.clear();
+            return Ok(());
+        }
+
+        revert_funcs.clear();
+        Ok(())
+    }
+
+    pub async fn drop_sink(&self, sink_id: SinkId) -> Result<()> {
+        let futures = self
+            .all_stream_clients()
+            .await?
+            .into_iter()
+            .map(|mut client| {
+                let request = ComputeNodeDropSinkRequest { sink_id };
+                async move { client.drop_sink(request).await.map_err(RwError::from) }
+            });
+        let _responses: Vec<_> = try_join_all(futures).await?;
+
+        Ok(())
+    }
+
+    async fn tick(&self) -> Result<()> {
```

Comment:
IIUC sink manager does not have to care about some change in downstream like redis/mysql etc. So it don't have to  monitor any changes.